### PR TITLE
Zooming and Scrolling

### PR DESCRIPTION
**Pull request by M320Trololol**
Title says it all, this really has it all: mouse scrolling (smooth), scroll wheel zoom, boundaries for the map, max/min zoom

### CODE REVIEW

**Comment by M320Trololol** [comment]
Review and merge asap, closes #57

File: client/assets/custommaps/default.tmx [hunk 1]
```diff
@@ -5,7 +5,7 @@
...
...VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVR8POiAAeQ==
```
Comment (ThoreKr) [inline client/assets/custommaps/default.tmx]: Binary in XML? srsly?

File: Rebleyama.iml [hunk 2]
```diff
@@ -1,10 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (danielrutz) [inline Rebleyama.iml]: Please do not commit project files. Remove this file and add it to .gitignore

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java [hunk 3]
```diff
@@ -55,8 +65,10 @@ public void render() {
         camera.update();
 		
 		//start up map renderer
```
Comment (danielrutz) [inline client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Indentation of this comment looks weird. Please use consistant code formatting.

Comment (danielrutz) [inline client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Indentation of this comment looks weird. Please use consistent code formatting.

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java [hunk 4]
```diff
@@ -66,22 +78,11 @@ public boolean keyDown(int keycode) {
 
     @Override
     public boolean keyUp(int keycode) {
```
Comment (danielrutz) [inline client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Methods that are intentionally left as stubs should be commented as such.

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java [hunk 5]
```diff
@@ -107,11 +108,81 @@ public boolean mouseMoved(int screenX, int screenY) {
...
 
 	@Override
 	public void dispose() {
 		// dispose of all the native resources
-	}
+    }
+    
+    private void handleMouseMovementInput() {
+        float mousePositionX = Gdx.input.getX(); 
+        float mousePositionY = Gdx.input.getY();
```
Comment (danielrutz) [inline client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Some comments about e.g. what the ifs do would really increase comprehensibility of the code.

File: Rebleyama.iml (hunk 2 above)
Comment (M320Trololol) [reply Rebleyama.iml]: Not my fault, fixed it anyway...

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java (hunk 3 above)
Comment (M320Trololol) [reply client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Fixed. Additional newline from merge conflict was the issue

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java (hunk 4 above)
Comment (M320Trololol) [reply client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Added javadoc comments to them to indicate what they'll be used for

File: client/src/main/java/de/rebleyama/client/RebleyamaClient.java (hunk 5 above)
Comment (M320Trololol) [reply client/src/main/java/de/rebleyama/client/RebleyamaClient.java]: Added comments, although the code is self-explanatory if you even look at it for a few seconds

**Pull request merged**